contrib/mesos/pkg/scheduler/podtask/predicate.go

```diff
@@ -57,21 +59,29 @@ func NodeSelectorPredicate(t *T, offer *mesos.Offer) bool {
 
 	// check the NodeSelector
 	if len(t.Pod.Spec.NodeSelector) > 0 {
-		slaveLabels := map[string]string{}
-		for _, a := range offer.Attributes {
-			if a.GetType() == mesos.Value_TEXT {
-				slaveLabels[a.GetName()] = a.GetText().GetValue()
+		l := map[string]string{
+			"kubernetes.io/hostname": offer.GetHostname(),
+		}
+		if n != nil && n.Labels != nil {
```

Comment:
Unnecessary `n.Labels != nil` check.
Re `n != nil`: this invariant is better maintained in the beginning of the func or elsewhere.